uvicorn/workers.py

```diff
@@ -76,9 +77,22 @@ def init_signals(self) -> None:
         # Don't let SIGUSR1 disturb active requests by interrupting system calls
         signal.siginterrupt(signal.SIGUSR1, False)
 
+    def _install_sigquit_handler(self) -> None:
+        """Workaround to install a SIGQUIT handler on workers.
+        - https://github.com/encode/uvicorn/issues/1116
+        - https://github.com/benoitc/gunicorn/issues/2604
+        """
+        if threading.current_thread() is not threading.main_thread():
+            # Signals can only be listened to from the main thread.
+            return
+
+        loop = asyncio.get_running_loop()
+        loop.add_signal_handler(signal.SIGQUIT, self.handle_exit, signal.SIGQUIT, None)
```

Comment:
Is all this logic necessary? Can't we add part of this in the `init_signals` method above?

Reply:
Maybe the question is misplaced because the signals are added to different processes? 🤔

Reply:
Seems like it, the signal handler needs to be attached to the main process's event loop, which will be available after `Worker.run()` calls `_serve()` inside an event loop (`asyncio.run()`).

The question is whether the thread check is necessary 🤔 because when a `SIGINT` signal is sent, the main thread will handle it anyways.